test/e2e/network_policy.go

```diff
@@ -0,0 +1,438 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/util/intstr"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+/*
+These Network Policy tests create two services A and B.
+There is a single pod in the service A , and a single pod on each node in
+the service B.
+
+Each pod is running a network monitor container (see test/images/network-monitor):
+-  The A pod uses service discovery to locate the B pods and waits
+   to establish communication (if expected) with those pods.
+-  Each B pod uses service discovery to locate the A pod and waits to
+   establish communication (if expected) with that pod.
+
+We test the following:
+-  Policy on, or off.  When policy is on we expect isolation between all pods across the
```

Comment:
"When policy is on" is a bit vague, I think this means to say "namespace ingress isolation is DefaultDeny".